substrate/windows/substrate-builder

```diff
@@ -0,0 +1,342 @@
+#!/usr/bin/env bash
+
+# Remove the defined logout command if possible to prevent
+# clearing the terminal when complete
+if [ -f /etc/bash.bash_logout ]; then
+    # NOTE: Don't exit on error here since it's only
+    # an attempt and failure does not affect the
+    # actual build
+    rm -f /etc/bash.bash_logout
+fi
+
+# Store substrates here
+output_dir="${1?Output directory is required}"
+# Find launchers here
+launcher_dir="${2?Launcher directory is required}"
+
+# Check for required tools and install if needed
+tools=( "git" "zip" "unzip" "python3" "curl" )
+needed=()
+
+for t in "${tools[@]}"; do
+    if ! command -v "${t}" > /dev/null; then
+        needed+=( "${t}" )
+    fi
+done
+
+if ! command -v dirname > /dev/null; then
+    needed+=( "coreutils" )
+fi
+
+if [ "${#needed[@]}" -gt 0 ]; then
+    pacman -S --noconfirm "${needed[@]}" || exit
```

Comment:
so weird to see pacman in this context! but lo and behold [that's what msys2 uses](https://www.msys2.org/docs/package-management/).